### PR TITLE
Log Payload on UnknownCloudFoundryException

### PR DESCRIPTION
**Pull request by ericbottard**
Fixes https://github.com/spring-cloud/spring-cloud-deployer-cloudfoundry/issues/175

### CODE REVIEW

File: src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java [hunk 1]
```diff
@@ -406,13 +403,9 @@ private String routePath(AppDeploymentRequest request) {
 				else if (isNotFoundError().test(error)) {
 					logger.warn("Unable to start application. It may have been destroyed before start completed: " + error.getMessage());
 				} else {
-					logger.error(String.format("Failed to start app %s", deploymentId), error);
+					logError(String.format("Failed to start app %s", deploymentId)).accept(error);
 				}
-
-			})
-			.otherwise(isNotFoundError(), e -> {logger.warn("Ignoring spurious 404 error during start: " + e.getMessage()); return Mono.empty();})
```
Comment (ilayaperumalg) [inline src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java]: What's the reason behind deleting these handlers?

Comment (ericbottard) [inline src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java]: Already taken care of above, if I'm not mistaken. I believe this is a previous merge error. @nebhale can confirm this is redundant

Comment (nebhale) [inline src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java]: Yep, looks like it's been taken care of by line 406/403 (although message differs).

Comment (markpollack) [inline src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java]: It looks like `doOnSuccess` should be added back though.

Comment (trisberg) [inline src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java]: `doOnSuccess` is logged on line 403 unless I'm reading it wrong

**Comment by trisberg** [comment]
LGTM

**Comment by trisberg** [comment]
Rebased and merged as commit e7802841c093575c4721efe70bd1ebb0013f6503

**Pull request closed without merging**